src/vm/amd64/JitHelpers_InlineGetThread.asm

```diff
@@ -147,6 +147,11 @@ align 16
         jmp     JIT_Box
 NESTED_END JIT_BoxFastMP_InlineGetThread, _TEXT
 
+RELATIVE_PTR_LOAD macro Reg, BaseReg, Offset
```

Comment:
Adding more instructions to allocation helpers is always unfortunate. 

Since you are touching this, it would be better to pass TemplateMT as the argument to the allocation helper from the JITed code. It would make the code even faster than what it is today.